### PR TITLE
sql: make SHOW JOBS more readable by humans

### PR DESCRIPTION
**Pull request by dt**
See commits.

Release note (breaking change): Some internal information columns have been removed from SHOW JOBS and job descriptions are abbreviated unless a specific job or set of jobs is shown (see prior release notes for details)

### CODE REVIEW

**Comment by dt** [comment]
I'm sure some tests are gonna complain but I'm going to let CI tell me which ones.

**Comment by dt** [comment]
next CI _should_ be green I think

**Review by stevendanna — APPROVED** [review]
TIL: `date_trunc`. 

We should maybe call this out as a breaking change.

File: pkg/sql/delegate/show_jobs.go [hunk 1]
```diff
@@ -22,12 +22,22 @@ import (
...
-	baseQuery.WriteString(`
-SELECT job_id, job_type, description, statement, user_name, status,
-       running_status, created, started, finished, modified,
-       fraction_completed, error, coordinator_id, trace_id, last_run,
-       next_run, num_runs, execution_errors
-`)
+	baseQuery.WriteString(`SELECT job_id, job_type, `)
+	if n.Jobs == nil {
+		baseQuery.WriteString(`CASE WHEN length(description) > 70 THEN `)
+		baseQuery.WriteString(`concat(substr(description, 0, 60)||' … '||right(description, 7)) `)
```
[the remark this answers is not in the document]
Comment (dt) [reply pkg/sql/delegate/show_jobs.go:28]: yeah, I think we already can have those here, so why not? also dots get used a lot so this is easy to match against

**Comment by dt** [comment]
I debated "breaking change" since we say that SHOW is _not_ stable and subject to exactly this kind of change since it is optimized for humans, but I've added one to the PR text for the merge commit.

**Pull request merged**